iOS/DittoChat/Models/Message.swift

```diff
@@ -20,12 +20,22 @@ struct Message: Identifiable, Equatable {
     var roomId: String
     var text: String
     var userId: String
-    var largeImageToken: DittoAttachmentToken?
-    var thumbnailImageToken: DittoAttachmentToken?
-    
+    var largeImageToken: [String: Any]?
+    var thumbnailImageToken: [String: Any]?
+
     var isImageMessage: Bool {
         thumbnailImageToken != nil || largeImageToken != nil
-    }    
+    }
+
+    // FIXME: Excluding attachment tokens from equality because Any is not equatable
+    static func == (lhs: Message, rhs: Message) -> Bool {
+        return lhs.id == rhs.id &&
+        lhs.createdOn == rhs.createdOn &&
+        lhs.roomId == rhs.roomId &&
+        lhs.text == rhs.text &&
+        lhs.userId == rhs.userId &&
+        lhs.isImageMessage == rhs.isImageMessage
+    }
```

Comment:
I've excluded the two token fields from equality so that `Message` can still conform to `Equatable`. If we can find a protocol to cover the 2nd dictionary value parameter what would allow us to use something more specific than `Any`. The values in these sub-dictionaries can be `Int`, `String` and `[String: String]` from what I've seen.